### PR TITLE
feat: switch to using httpget for liveness-probe

### LINKED ISSUE #452: use healthz and readyz endpoints instead of exec probes
**Describe the solution you'd like**
Instead of calling grpc_health_probe client to check the socket, we should expose /healthz and /readyz endpoints

We can also remove this extra binary after this is done

https://github.com/Azure/secrets-store-csi-driver-provider-azure/blob/master/deployment/provider-azure-installer.yaml#L37-L44

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Secrets Store CSI Driver version: (use the image tag):
- Azure Key Vault provider version: (use the image tag):
- Kubernetes version: (use `kubectl version`):
- Cluster type: (e.g. AKS, aks-engine, etc):

### PR DESCRIPTION
**Pull request by aramase**
Signed-off-by: Anish Ramasekar <anish.ramasekar@gmail.com>

**Reason for Change**:

- Switches to using http probe for `healthz`

**Requirements**

- [x] squashed commits
- [x] included documentation
- [x] added unit tests and e2e tests (if applicable).

**Issue Fixed**:

fixes #452 

**Does this change contain code from or inspired by another project?**

- [ ] Yes
- [x] No

**If "Yes," did you notify that project's maintainers and provide attribution?**

**Special Notes for Reviewers**:

### CODE REVIEW

File: manifest_staging/charts/csi-secrets-store-provider-azure/templates/provider-azure-installer.yaml [hunk 1]
```diff
@@ -46,19 +46,17 @@ spec:
             {{- if .Values.linux.customUserAgent }}
             - --custom-user-agent={{ .Values.linux.customUserAgent }}
             {{- end }}
-          lifecycle:
```
Comment (aramase) [inline manifest_staging/charts/csi-secrets-store-provider-azure/templates/provider-azure-installer.yaml:49]: Removing the prestop hook as the socket is cleaned up during provider startup

File: manifest_staging/deployment/provider-azure-installer.yaml [hunk 2]
```diff
@@ -29,19 +29,17 @@ spec:
           args:
             - --endpoint=unix:///provider/azure.sock
             - --construct-pem-chain=false
-          lifecycle:
-            preStop:
-              exec:
-                command:
-                  - "rm /provider/azure.sock"
-          readinessProbe:
```
Comment (sozercan) [inline manifest_staging/deployment/provider-azure-installer.yaml:37]: do we want to remove `readinessProbe`?

Comment (aramase) [reply manifest_staging/deployment/provider-azure-installer.yaml:37]: `readinessProbes` are typically used by kubelet to determine if the pod can receive traffic. That really doesn't apply for the driver-provider communication. As long as the socket exists, the driver will make a RPC call to the provider.

**Review by sozercan — APPROVED** [review]
LGTM

**Pull request merged**